pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/DeletePatchFactory.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.dsm.model.patch;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.NonNull;
+import org.broadinstitute.dsm.db.OncHistoryDetail;
+import org.broadinstitute.dsm.db.dao.ddp.tissue.TissueDao;
+import org.broadinstitute.dsm.db.dao.ddp.tissue.TissueSMIDDao;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.model.NameValue;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.util.NotificationUtil;
+
+public class DeletePatchFactory {
+
+    private static final String ONC_HISTORY_DELETED_FIELD = "oD.deleted";
+    private static final String TISSUE_DELETED_FIELD = "t.deleted";
+    private static final String SM_ID_DELETED_FIELD = "sm.deleted";
+    private static final String TRUE_FLAG = "1";
+
+    public static BasePatch produce(Patch patch, NotificationUtil notificationUtil) {
+        BasePatch patcher;
+        if (PatchFactory.isOncHistoryDetailPatch(patch)) {
+            patcher = new DeletePatch(patch, notificationUtil, DeleteType.ONC_HISTORY_DETAIL);
+        } else if (PatchFactory.isTissueRelatedOncHistoryId(patch)) {
+            patcher = new DeletePatch(patch, notificationUtil, DeleteType.TISSUE);
+        }  else if (PatchFactory.isSmIdPatch(patch)) {
+            patcher = new DeletePatch(patch, notificationUtil, DeleteType.SM_ID);
+        } else {
+            throw new DsmInternalError("This method should not reach here");
+        }
+        return patcher;
+    }
+
+    protected static void deleteChildrenFields(@NonNull Patch originalPatch, NotificationUtil notificationUtil) {
+        List<Patch> deletePatches = new ArrayList<>();
+        if (isTissuePatch(originalPatch)) {
+            deletePatches = getPatchForSmIds(originalPatch);
+        } else if (isOncHistoryPatch(originalPatch)) {
+            deletePatches = getPatchForTissues(originalPatch);
+        }
+        for (Patch childPatch : deletePatches) {
+            BasePatch patcher = PatchFactory.makePatch(childPatch, notificationUtil);
+            patcher.doPatch();
+        }
+    }
+
+    private static boolean isOncHistoryPatch(Patch originalPatch) {
+        return originalPatch.getNameValue().getName().equals(ONC_HISTORY_DELETED_FIELD) &&
```

Comment:
Consider moving these methods into `Patch` class itself so that clients can just ask `Patch`: are you an onc history patch?  And then Patch itself can have control over how it answers the question.